src/management-system-v2/components/confirmation-button.tsx

```diff
@@ -0,0 +1,77 @@
+import { ComponentProps, FC, PropsWithChildren, useState } from 'react';
+import { Button, Modal, Tooltip } from 'antd';
+
+type ConfirmationModalProps = {
+  onConfirm: (...args: any[]) => Promise<any> | any;
+  title: string;
+  description: string;
+  canCloseWhileLoading?: boolean;
+  modalProps?: Omit<
+    ComponentProps<typeof Modal>,
+    'open' | 'title' | 'onOk' | 'confirmLoading' | 'onCancel'
+  >;
+  buttonProps?: ComponentProps<typeof Button>;
+  tooltip?: string;
+};
+
+const ConfirmationButton: FC<PropsWithChildren<ConfirmationModalProps>> = ({
+  children,
+  onConfirm,
+  title,
+  description,
+  canCloseWhileLoading = false,
+  modalProps,
+  buttonProps,
+  tooltip,
+}) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const clearModal = () => {
+    setModalOpen(false);
+    setLoading(false);
+  };
+
+  const onConfirmWrapper = () => {
```

Comment:
Is there a reason to do it like this instead of:
```
const onConfirmWrapper = async () => {
  setLoading(true);
  
  try {
    await onConfirm();
  } catch (err) {}
  
  clearModal();
}
```

There might be a tiny bit more overhead for functions that do not return a promise but it is less convoluted i think.